python/shotgun_data/shotgun_data_retriever.py

```diff
@@ -689,14 +829,23 @@ def _get_thumbnail_path(url, bundle, directory_only=False):
         # If we were only asked to give back a directory path then we can
         # skip building and appending a file name.
         if not directory_only:
-            path_chunks.append("%s.jpeg" % hash_str[4:])
+            path_chunks.append(
+                "%s%s" % (hash_str[4:], ".jpeg" if use_default_extension else "")
```

Comment:
hmm, looking at both the new and the old code, what happens when `directory_only` is `True`? Looks like `hash_str[4:]` is dropped which cannot be correct?